---
title: eslint-cdk-plugin - pascal-case-construct-id
titleTemplate: ":title"
---

# pascal-case-construct-id

<div class="info-item">
  ‚úÖ Using
  <a href="/rules/#recommended-rules">recommended</a>
  in an ESLint configuration enables this rule.
</div>

<div class="info-item">
  üîß Some problems reported by this rule are automatically fixable by the
  <a href="https://eslint.org/docs/latest/use/command-line-interface#--fix">
    --fix ESLint command line option
  </a>
</div>

This rule enforces writing Construct IDs in PascalCase.

Enforcing a consistent naming convention helps developers manage logical IDs more easily, and as a result, helps reduce the risk of unintentional logical ID collisions.

(This rule applies only to classes that extends from `Construct` or `Stack`.)

---

#### üîß How to use

```js
// eslint.config.mjs
export default [
  {
    // ... some configs
    rules: {
      "cdk/pascal-case-construct-id": "error",
    },
  },
];
```

#### ‚úÖ Correct Example

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";

// ‚úÖ Can use PascalCase
const bucket = new Bucket(this, "MyBucket");
```

#### ‚ùå Incorrect Example

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";

// ‚ùå Shouldn't use camelCase
const bucket = new Bucket(this, "myBucket");

// ‚ùå Shouldn't use snake_case
const bucket = new Bucket(this, "my_bucket");

// ‚ùå Shouldn't use kebab-case
const bucket = new Bucket(this, "my-bucket");
```
