---
title: eslint-plugin-awscdk - require-passing-this
titleTemplate: ":title"
---

<script setup>
import RecommendedItem from '../../components/RecommendedItem.vue'
import FixableItem from '../../components/FixableItem.vue'
import Playground from '../../components/Playground.vue'
</script>

# require-passing-this

<RecommendedItem japanese />
<FixableItem japanese />

ã“ã®ãƒ«ãƒ¼ãƒ«ã¯ `Construct` ã®ã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã« `this` ã‚’æ¸¡ã™ã“ã¨ã‚’å¼·åˆ¶ã—ã¾ã™ã€‚

AWS CDK ãƒªã‚½ãƒ¼ã‚¹ã‚’ä½œæˆã™ã‚‹ã¨ãã€`Construct` ã« `this` ã‚’æ¸¡ã™ã“ã¨ã¯æ­£ã—ã„ãƒªã‚½ãƒ¼ã‚¹éšå±¤ã‚’ç¶­æŒã™ã‚‹ãŸã‚ã«é‡è¦ã§ã™ã€‚

Construct ã®ã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã®ç¬¬ä¸€å¼•æ•°ã¸ `this` ä»¥å¤–ã®å€¤ (ç‰¹ã«ã€è¦ªã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã‹ã‚‰å—ã‘å–ã£ãŸ `scope` å¤‰æ•°ãªã©) ã‚’æ¸¡ã—ã¦ã—ã¾ã†ã¨ã€æ¬¡ã®ã‚ˆã†ãªå•é¡ŒãŒç™ºç”Ÿã™ã‚‹å¯èƒ½æ€§ãŒã‚ã‚Šã¾ã™

- ç”Ÿæˆã•ã‚Œã‚‹ CloudFormation ãƒ†ãƒ³ãƒ—ãƒ¬ãƒ¼ãƒˆã®ãƒªã‚½ãƒ¼ã‚¹éšå±¤ãŒæ­£ã—ããªã„
- äºˆæœŸã—ãªã„ãƒªã‚½ãƒ¼ã‚¹ã®å‘½å

(ã“ã®ãƒ«ãƒ¼ãƒ«ã¯ `Construct` ã‹ã‚‰æ´¾ç”Ÿã—ãŸã‚¯ãƒ©ã‚¹ã«ã®ã¿é©ç”¨ã•ã‚Œã¾ã™)

---

#### ğŸ”§ ä½¿ç”¨æ–¹æ³•

```js
// eslint.config.mjs
export default defineConfig([
  {
    // ... some configs
    rules: {
      "cdk/require-passing-this": "error",
    },
  },
]);
```

#### âœ… æ­£ã—ã„ä¾‹

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    const sample = new SampleConstruct(this, "Sample");

    // âœ… `this` ã¯å¸¸ã«ä½¿ç”¨ã§ãã¾ã™
    new Bucket(this, "SampleBucket");

    // âœ… `sample` (Construct ã®ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹) ã‚’ã‚¹ã‚³ãƒ¼ãƒ—ã¨ã—ã¦æ¸¡ã™ã“ã¨ãŒè¨±å¯ã•ã‚Œã‚‹
    new OtherConstruct(sample, "Child");
  }
}
```

#### âŒ ä¸æ­£ãªä¾‹

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // âŒ scope ã‚’ä½¿ç”¨ã™ã¹ãã§ã¯ã‚ã‚Šã¾ã›ã‚“
    new Bucket(scope, "SampleBucket");
  }
}
```

## ã‚ªãƒ—ã‚·ãƒ§ãƒ³

```ts
type Options = {
  allowNonThisAndDisallowScope: boolean;
};

const defaultOptions: Options = {
  allowNonThisAndDisallowScope: true,
};
```

### `allowNonThisAndDisallowScope`

Construct ã®ã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã®ç¬¬ä¸€å¼•æ•° (ã‚¹ã‚³ãƒ¼ãƒ—) ã¨ã—ã¦ã€`this` ä»¥å¤–ã®å€¤ã‚’è¨±å¯ã™ã‚‹ã‹ã©ã†ã‹ã‚’æ±ºå®šã—ã¾ã™ã€‚

- `false`: æ–°ã—ã„ Construct ã‚’ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹åŒ–ã™ã‚‹éš›ã€ç¬¬ä¸€å¼•æ•° (ã‚¹ã‚³ãƒ¼ãƒ—) ã¨ã—ã¦ `this` ã®ã¿ãŒè¨±å¯ã•ã‚Œã¾ã™
- `true`: `this` ä»¥å¤–ã® Construct ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹ã‚’ç¬¬ä¸€å¼•æ•° (ã‚¹ã‚³ãƒ¼ãƒ—) ã¨ã—ã¦æ¸¡ã™ã“ã¨ã‚’è¨±å¯ã—ã¾ã™
  - ãŸã ã—ã€è¦ªã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ãŒå—ã‘å–ã£ãŸ `scope` å¤‰æ•°ã‚’ç›´æ¥ä½¿ç”¨ã™ã‚‹ã“ã¨ã¯å¼•ãç¶šãç¦æ­¢ã•ã‚Œã¾ã™
  - ã“ã®è¨­å®šã¯ã€ãƒã‚¹ãƒˆã•ã‚ŒãŸ Construct éšå±¤ã‚’ä½œæˆã™ã‚‹å ´åˆã«ä¾¿åˆ©ã§ã™ã€‚

#### âœ… Correct Example

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // âœ… `this` ã¯å¸¸ã«ä½¿ç”¨ã§ãã¾ã™
    new Bucket(this, "SampleBucket");
  }
}
```

#### âŒ Incorrect Example

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    const sample = new SampleConstruct(this, "Sample");

    // âŒ scope ã‚’ä½¿ç”¨ã™ã¹ãã§ã¯ã‚ã‚Šã¾ã›ã‚“ã€
    new Bucket(scope, "SampleBucket");

    // âŒ ä»–ã® Construct ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹ã‚’ã‚¹ã‚³ãƒ¼ãƒ—ã¨ã—ã¦ä½¿ç”¨ã™ã¹ãã§ã¯ã‚ã‚Šã¾ã›ã‚“ã€‚
    new OtherConstruct(sample, "Child");
  }
}
```

<Playground link="https://eslint-online-playground.netlify.app/#eNqNVc1y0zAQfpUdX0g6+WHKLQwzQMuRliHlVHfGqr1J1CiSkeSkmUyO3HgEeDmehF35N2layMGOtd/ufvurXeRdavRMzkcPzuhoEu1iDRBHqVnlUqG9zr002sXRBIKEZV7YOXo+iqNP0/PX5+dxNKiFSt6z5LYSvYmju0a2MlmhsFa8wkcyciT8is6ogn2WsPtCZ0Sjg3PeyjR497bA5tga4y8lAVnL2bTS2NNzH+towGdji98LaXGYC+ekng/9QrqRdxS2XOXGetjBBUVLhlMPe5hZswq5qI4oDW9j3UA/FukSOzixccM0Ww4pBWP+7yh4wscaH4NGqsgtfN62LigDqDPXcRqS3Dg0tkflyXHSIgYgswlwEvS8X9fEFTlWUJb3g1cWjMfw59cP+MbRQsLhJiAcBCRIB0JtxJZfymwwG5VKGjdVbD3WGFBoU7HKFZaHcXRo//dPmC5MoTL9ykPhEPwCIRcWtX/lIAm+ElgLK8W9qgtGivRj80xD6rVQMgNco4bNgh6B0ZXRN4T4oLNL6cLJtCbOtX9Kt8rAKb7UCqETYk2ub4jhzLBBzgs+BjSbLWkEBslLFBIGJ0wigR4lVOoQtUWamxWVFDNw6PvEMNQSXOnhXSDb6YCjBJdUq6IlpRI70OTAeaFTBDMD0XZDvyxiqF5TWPLazUmwwj4uFlJl7IKmAZ2S2o+a2W9HoJTUPf2+/Bw/HPV+hjOpkXiQejsBFbi02lXw7tCs3+boUitzPywFXTDN0BdVzCmnB2bDbOVBcDhXxEUUKrwbTr1bLvlBlG7UKU9ow9GIhv8fmGrCZrQMXVhrZ2fjM1Jr91o1xCRsiJ+yVsMtbTkCV3bDJLzUaROYCeVw0oFfa7Wth/l0+RsoLa9seXLtlUu6RfIORWtNu2qr493/0At7tlGqQy0P+XkXmi4X6VLM8eiq4V4o13Z9PwS1OMpwfYk5J0+nEo/uoKcNxBaU8Oi6t0qLewZwordewD0j7Gz+ZxDdO6QLqBdTtP8L/eKMjg==" />
