---
title: eslint-plugin-awscdk - require-passing-this
titleTemplate: ":title"
---

<script setup>
import RecommendedItem from '../../components/RecommendedItem.vue'
import FixableItem from '../../components/FixableItem.vue'
import Playground from '../../components/Playground.vue'
</script>

# require-passing-this

<RecommendedItem japanese />
<FixableItem japanese />

ã“ã®ãƒ«ãƒ¼ãƒ«ã¯ `Construct` ã®ã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã« `this` ã‚’æ¸¡ã™ã“ã¨ã‚’å¼·åˆ¶ã—ã¾ã™ã€‚

AWS CDK ãƒªã‚½ãƒ¼ã‚¹ã‚’ä½œæˆã™ã‚‹ã¨ãã€`Construct` ã« `this` ã‚’æ¸¡ã™ã“ã¨ã¯æ­£ã—ã„ãƒªã‚½ãƒ¼ã‚¹éšå±¤ã‚’ç¶­æŒã™ã‚‹ãŸã‚ã«é‡è¦ã§ã™ã€‚

Construct ã®ã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã®ç¬¬ä¸€å¼•æ•°ã¸ `this` ä»¥å¤–ã®å€¤ (ç‰¹ã«ã€è¦ªã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã‹ã‚‰å—ã‘å–ã£ãŸ `scope` å¤‰æ•°ãªã©) ã‚’æ¸¡ã—ã¦ã—ã¾ã†ã¨ã€æ¬¡ã®ã‚ˆã†ãªå•é¡ŒãŒç™ºç”Ÿã™ã‚‹å¯èƒ½æ€§ãŒã‚ã‚Šã¾ã™

- ç”Ÿæˆã•ã‚Œã‚‹ CloudFormation ãƒ†ãƒ³ãƒ—ãƒ¬ãƒ¼ãƒˆã®ãƒªã‚½ãƒ¼ã‚¹éšå±¤ãŒæ­£ã—ããªã„
- äºˆæœŸã—ãªã„ãƒªã‚½ãƒ¼ã‚¹ã®å‘½å

(ã“ã®ãƒ«ãƒ¼ãƒ«ã¯ `Construct` ã‹ã‚‰æ´¾ç”Ÿã—ãŸã‚¯ãƒ©ã‚¹ã«ã®ã¿é©ç”¨ã•ã‚Œã¾ã™)

---

#### ğŸ”§ ä½¿ç”¨æ–¹æ³•

```js
// eslint.config.mjs
export default defineConfig([
  {
    // ... some configs
    rules: {
      "awscdk/require-passing-this": "error",
    },
  },
]);
```

#### âœ… æ­£ã—ã„ä¾‹

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    const sample = new SampleConstruct(this, "Sample");

    // âœ… `this` ã¯å¸¸ã«ä½¿ç”¨ã§ãã¾ã™
    new Bucket(this, "SampleBucket");

    // âœ… `sample` (Construct ã®ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹) ã‚’ã‚¹ã‚³ãƒ¼ãƒ—ã¨ã—ã¦æ¸¡ã™ã“ã¨ãŒè¨±å¯ã•ã‚Œã‚‹
    new OtherConstruct(sample, "Child");
  }
}
```

#### âŒ ä¸æ­£ãªä¾‹

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // âŒ scope ã‚’ä½¿ç”¨ã™ã¹ãã§ã¯ã‚ã‚Šã¾ã›ã‚“
    new Bucket(scope, "SampleBucket");
  }
}
```

## ã‚ªãƒ—ã‚·ãƒ§ãƒ³

```ts
type Options = {
  allowNonThisAndDisallowScope: boolean;
};

const defaultOptions: Options = {
  allowNonThisAndDisallowScope: true,
};
```

### `allowNonThisAndDisallowScope`

Construct ã®ã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ã®ç¬¬ä¸€å¼•æ•° (ã‚¹ã‚³ãƒ¼ãƒ—) ã¨ã—ã¦ã€`this` ä»¥å¤–ã®å€¤ã‚’è¨±å¯ã™ã‚‹ã‹ã©ã†ã‹ã‚’æ±ºå®šã—ã¾ã™ã€‚

- `false`: æ–°ã—ã„ Construct ã‚’ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹åŒ–ã™ã‚‹éš›ã€ç¬¬ä¸€å¼•æ•° (ã‚¹ã‚³ãƒ¼ãƒ—) ã¨ã—ã¦ `this` ã®ã¿ãŒè¨±å¯ã•ã‚Œã¾ã™
- `true`: `this` ä»¥å¤–ã® Construct ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹ã‚’ç¬¬ä¸€å¼•æ•° (ã‚¹ã‚³ãƒ¼ãƒ—) ã¨ã—ã¦æ¸¡ã™ã“ã¨ã‚’è¨±å¯ã—ã¾ã™
  - ãŸã ã—ã€è¦ªã‚³ãƒ³ã‚¹ãƒˆãƒ©ã‚¯ã‚¿ãŒå—ã‘å–ã£ãŸ `scope` å¤‰æ•°ã‚’ç›´æ¥ä½¿ç”¨ã™ã‚‹ã“ã¨ã¯å¼•ãç¶šãç¦æ­¢ã•ã‚Œã¾ã™
  - ã“ã®è¨­å®šã¯ã€ãƒã‚¹ãƒˆã•ã‚ŒãŸ Construct éšå±¤ã‚’ä½œæˆã™ã‚‹å ´åˆã«ä¾¿åˆ©ã§ã™ã€‚

#### âœ… Correct Example

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // âœ… `this` ã¯å¸¸ã«ä½¿ç”¨ã§ãã¾ã™
    new Bucket(this, "SampleBucket");
  }
}
```

#### âŒ Incorrect Example

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    const sample = new SampleConstruct(this, "Sample");

    // âŒ scope ã‚’ä½¿ç”¨ã™ã¹ãã§ã¯ã‚ã‚Šã¾ã›ã‚“ã€
    new Bucket(scope, "SampleBucket");

    // âŒ ä»–ã® Construct ã‚¤ãƒ³ã‚¹ã‚¿ãƒ³ã‚¹ã‚’ã‚¹ã‚³ãƒ¼ãƒ—ã¨ã—ã¦ä½¿ç”¨ã™ã¹ãã§ã¯ã‚ã‚Šã¾ã›ã‚“ã€‚
    new OtherConstruct(sample, "Child");
  }
}
```

<Playground link="https://eslint-online-playground.netlify.app/#eNqNVc1uEzEQfpXRXppU+UHlFoQEtBxpESmnbqV1dyeJW8debG/SKMqRG48AL8eTMOP9TZqW9pA0M9/MfPPrbeRdavRMzkf3zuhoEm1jDRBHqVnmUqG9yr002sXRBIKGdV7YOXoWxdHn6dmbs7M4GtRKJe9Yc1Op3sbRbaNbmqxQWBte4iM5OVB+Q2dUwTFL2F2hM6LRwTlvZRqie1tgI7bG+AtJQLZyNq0sdvS5i3U0YNnY4o9CWhzmwjmp50O/kG7kHaUtl7mxHrZwTtmS49TDDmbWLEMtKhGV4V2sG+inIn3ADk6s3TDNHoZUgjH/7yh5wscaH4NFqigsfNm0IagCqDPXCRqK3AQ0tkftyXHSIgYgswlwEfS8X/fEFTlWUNb3Q1RWjMfw9/dP+M7ZQsLpJiAcBCRIB0KtxYa/lFljNiqNNK6r3HpsMaDUpmKZKyyFcbTv/88vmC5MoTJ94qFwCH6BkAuL2p84SEKsBFbCSnGn6oaRIf2xe6Yh9UoomQGuUMN6QR+B0aXR14T4qLML6YJkWhPn3j+lW1XgGF8ahTAJsabQ18RwZtgh1wUfA5rdljQCg+QlCgmDEyaRQI8KKnXI2iLtzZJaihk49H1iGHoJrozwPpDtTMBBgUuqVdOS0ogDaArgvNApgpmBaKehXzYxdK9pLEXt1iR44RjnC6kyDkHbgE5J7UfN7rcrUGrqmf5Q/hzfH8x+hjOpkXiQebsBFbj02jXwbt+t3+ToUitzPywVXTDt0FdVzKmme26HeRAOabUIsb9aREcUKnw3tHo33PW9RN2o06EwiaMR7f9/MNWSzegeunDZTk/Hp2TWnrZqj0nZcD/mrYZbOnQErvyGZXhp2CYwE8rhpAO/0mpT7/PxCWig4S4RraPHrzzVLZgvKVpr2oNbibevYRiubWNUZ1sK+fM2jF4u0gcxx4MHhyeiPN71KxHM4ijD1QXmXD+dSjx4iZ6OEXtQwqPrvi0t7hnA8Ql7AfqMsvMEPIPoPiZdQH2hot0/q6uQbw==" />
